share/functions/fish_clipboard_copy.fish

```diff
@@ -1,7 +1,16 @@
 function fish_clipboard_copy
-    # Copy the current selection, or the entire commandline if that is empty.
-    set -l cmdline (commandline --current-selection | string collect)
-    test -n "$cmdline"; or set cmdline (commandline | string collect)
+    set -l cmdline
+    if isatty stdin
+        # Copy the current selection, or the entire commandline if that is empty.
+        set cmdline (commandline --current-selection | string collect)
+        test -n "$cmdline"; or set cmdline (commandline | string collect)
+    else
+        # Read from stdin
+        while read -lz line
+            set -a cmdline $line
```

Comment:
That's a neat way to losslessly forward data; notablky it works if stdin does not have a trailing new line.
We should use it for the other as well when we are not reading from stdin. I can take care of that.

Reply:
The other case can just be done via `string collect -N`, if you want to care about not removing trailing newlines (we already don't *add* any).

(we could also use that here if we had a `--stdin` flag to force `string` to read from stdin)

Reply:
Nice, `string collect -N` does the job!